src/ui/render.ts

```diff
@@ -278,39 +293,82 @@ export default class UI {
 
   private registerEvents() {
     this.lightSheet.events.on(EventType.CORE_SET_CELL, (event) => {
-      if (this.lightSheet.isReady) this.onCoreSetCell(event);
+      this.onCoreSetCell(event);
     });
   }
 
   private onCoreSetCell(event: LightsheetEvent) {
     const payload = event.payload as CoreSetCellPayload;
-    // Get HTML elements and (new) IDs for the payload's cell and row.
-    const elInfo = LightSheetHelper.getElementInfoForSetCell(payload);
-
-    if (!elInfo.rowDom) {
-      const row = this.addRow(payload.indexPosition.row);
-      elInfo.rowDom = row;
-      row.id = elInfo.rowDomId;
+    // Create new columns if the column index is greater than the current column count.
+    const newColumns = payload.indexPosition.column - this.getColumnCount() + 1;
+    for (let i = 0; i < newColumns; i++) {
+      this.addColumn();
     }
-    if (!elInfo.cellDom) {
-      elInfo.cellDom = this.addCell(
-        elInfo.rowDom!,
-        payload.indexPosition.column,
-        payload.indexPosition.row,
-        payload.formattedValue,
-        payload.keyPosition.columnKey?.toString(),
-      );
+
+    const newRows = payload.indexPosition.row - this.getRowCount() + 1;
+    for (let i = 0; i < newRows; i++) {
+      this.addRow();
     }
 
-    elInfo.cellDom.id = elInfo.cellDomId;
-    elInfo.rowDom.id = elInfo.rowDomId;
+    // Get HTML elements and (new) IDs for the payload's cell and row.
+    const elInfo = this.getElementInfoForSetCell(payload);
+
+    elInfo.cellDom!.id = elInfo.cellDomId;
+    elInfo.rowDom!.id = elInfo.rowDomId;
 
     // Set cell value to resolved value from the core.
     // TODO Cell formula should be preserved. (Issue #49)
-    (elInfo.cellDom.firstChild! as HTMLInputElement).value =
+    (elInfo.cellDom!.firstChild! as HTMLInputElement).value =
       payload.formattedValue;
   }
 
+  private getElementInfoForSetCell = (payload: CoreSetCellPayload) => {
+    const colKey = payload.keyPosition.columnKey?.toString();
+    const rowKey = payload.keyPosition.rowKey?.toString();
+
+    const columnIndex = payload.indexPosition.column;
+    const rowIndex = payload.indexPosition.row;
+
+    const cellDomKey =
+      colKey && rowKey ? `${colKey!.toString()}_${rowKey!.toString()}` : null;
+
+    // Get the cell by either column and row key or position.
+    // TODO Index-based ID may not be unique if there are multiple sheets.
+    const cellDom =
+      (cellDomKey && document.getElementById(cellDomKey)) ||
+      document.getElementById(`${columnIndex}_${rowIndex}`);
```

Comment:
This can be simplified as well :^)

Reply:
This is also not equivalent

Reply:
My bad, this was as concise as it can be.